### PR TITLE
Re-add unit test, update text about the snippet, and fix older tests.

### PR DESCRIPTION
**Pull request by MongoCaleb**
## Pull Request Info - SDK Docs Consolidation

Jira ticket: [DOCSP-39224](https://jira.mongodb.org/browse/DOCSP-39224)

*Staged Page*

- [Check Upload and Download Progress](https://deploy-preview-3317--device-sdk.netlify.app/sdk/dotnet/sync/sync-progress/#monitor-sync-progress)

### CODE REVIEW

**Review by cbullinger — APPROVED** [review]
Some small fixes and suggestions

File: source/sdk/dotnet/sync/sync-progress.txt [hunk 1]
```diff
@@ -81,15 +81,14 @@ bytes to be transferred.
...
-   observable on the ``session`` to listen for upload events indefinitely. When 
-   this callback is triggered, it prints the number of transferred 
-   bytes and the number of transferable bytes to the console.
+   observable on the ``session`` to listen for upload events. When 
+   this callback is triggered, it prints the upload progress as a percentage.
 
    .. literalinclude:: /examples/generated/dotnet/ProgressNotifications.snippet.upload-download-progress-notification.cs
       :language: csharp
 
    Once you no longer wish to receive notifications, unregister the token by using
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/sdk/dotnet/sync/sync-progress.txt:90]: Not sure about the first suggestion: I think it needs to come after the info on subscribing. 
Agreed on 2nd bullet.

File: source/sdk/dotnet/sync/sync-progress.txt [hunk 2]
```diff
@@ -81,15 +81,14 @@ bytes to be transferred.
 .. example::
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/sdk/dotnet/sync/sync-progress.txt:75]: I chose to remove the notes and update the text above to be clear that this new value is an estimate. I think the details previously provided are not longer applicable and were confusing.

File: source/sdk/dotnet/sync/sync-progress.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/sdk/dotnet/sync/sync-progress.txt:84]: ah, I must've skimmed too quickly. Based on the [docs](https://www.mongodb.com/docs/realm-sdks/dotnet/latest/reference/Realms.Sync.ProgressMode.html#:~:text=or%20until%20it%20is%20unregistered%20by%20disposing%20the%20subscription%20token.), I thought it was only applicable to the `ReportIndefinitely` mode

File: source/sdk/dotnet/sync/sync-progress.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/sdk/dotnet/sync/sync-progress.txt:75]: sgtm! I wasn't familiar enough to easily tell whether any of it was still applicable 👍

**Pull request merged**